packages/cardano-services/src/Program/services/ogmios.ts

```diff
@@ -163,6 +169,25 @@ export const ogmiosCardanoNodeWithDiscovery = async (
   return Object.setPrototypeOf(cardanoNodeProxy, RunnableModule.prototype);
 };
 
+/**
+ * Creates an ObservableOgmiosCardanoNode instance :
+ * - use passed srv service name in order to resolve the port
+ * - all other operations are bind to pool object without modifications
+ *
+ * @returns ObservableOgmiosCardanoNode instance
+ */
+export const ogmiosObservableCardanoNodeWithDiscovery = async (
+  dnsResolver: DnsResolver,
+  logger: Logger,
+  serviceName: string
+): Promise<OgmiosObservableCardanoNode> =>
+  new OgmiosObservableCardanoNode(
+    {
+      connectionConfig$: from(dnsResolver(serviceName).then(({ name, port }) => ({ host: name, port })))
```

Comment:
Each subscription must re-resolve
```suggestion
      connectionConfig$: defer(() => from(dnsResolver(serviceName).then(({ name, port }) => ({ host: name, port }))))
```